plugins/modules/azure_rm_monitoractivitylogalerts.py

```diff
@@ -0,0 +1,514 @@
+                if update_tags or not self.default_compare({}, activity_log_alert_update, before_dict, '', result_compare):
+                    activity_log_alert_update['tags'] = update_tags_content
+                    # Need to create/update the activity log alert; changed -> True
+                    self.results['changed'] = True
+                    if self.check_mode:
+                        # Check mode, skipping actual creation
+                        pass
+                    else:
+                        create_response = self.create_or_update(activity_log_alert_update)
+
+            if self.check_mode or not self.results['changed']:
+                # When object was not updated or when running in check mode
+                # assume activity_log_alert_update is resulting object
+                result = activity_log_alert_update
+            else:
+                # otherwise take resulting new object from response of create call
+                result = create_response
+
+        # Delete activity log alert if state is absent and it exists
+        # if it doesn't exist, it's already absent
+        elif self.state == 'absent' and before_dict is not None:
+            self.results['changed'] = True
+            if self.check_mode:
+                # do not delete in check mode
+                pass
+            else:
+                self.delete()
+
+        self.results['diff']['before'] = before_dict
+        self.results['diff']['after'] = result
+        self.results['activitylogalert'] = result
+
+        return self.results
+
+    def get_activity_log_alert(self):
+        '''
+        Gets the properties of the specified activity log alert.
+
+        :return: List of Activity Log Alerts
+        '''
+        self.log("Checking if activity log alert {0} in resource group {1} is present".format(self.name,
+                                                                                              self.resource_group))
+
+        result = None
+        response = None
+
+        try:
+            response = self.monitor_management_client_activity_log_alerts.activity_log_alerts.get(activity_log_alert_name=self.name,
+                                                                                                  resource_group_name=self.resource_group)
+        except ResourceNotFoundError as ex:
+            self.log("Could not find activity log alert {0} in resource group {1}".format(self.name, self.resource_group))
+        if response:
+            result = self.serialize_obj(response, AZURE_OBJECT_CLASS)
+
+        return result
+
+    def create_or_update(self, activity_log_alert_update):
+        result = None
+        response = None
+        activity_log_alerts = self.monitor_management_client_activity_log_alerts.activity_log_alerts
+        try:
+            response = activity_log_alerts.create_or_update(resource_group_name=self.resource_group,
```

Comment:
@Klaas- This module related python-sdk provides an update method(https://learn.microsoft.com/en-us/python/api/azure-mgmt-monitor/azure.mgmt.monitor.v2020_10_01.aio.operations.activitylogalertsoperations?view=azure-python#azure-mgmt-monitor-v2020-10-01-aio-operations-activitylogalertsoperations-update). Why not use this update method? The current approach is very thorough, overwriting the original configuration with the new one. Thank you!